### PR TITLE
Velocity analysis

### PR DESCRIPTION
**Pull request by matt-gust**
Added functions for velocity analysis as well as a vignette showing how these functions can be used in conjuction with scVelo in python

### CODE REVIEW

File: inst/make_loom_files.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+env_path=placeholder_env_path
+gtf_file=placeholder_gtf_file
+sampleid=placeholder_sampleid
+
+#Check if conda environment exists
+if conda info --envs | grep -q ${env_path};
+then
+    echo "env already exists";
+else
+    conda create -p ${env_path} -f make_environment.yml
```
Comment (MVesuviusC) [inline inst/make_loom_files.sh]: I think you may need to specify the full path to the make_environment.yml file. This likely works when you test it, but when someone else installs the package that file won't be in whatever folder they're working from. You can get the location of the file similar to I do here: https://github.com/kidcancerlab/rrrSnvs/blob/cf535ec028815b5edacf91ca7c985830e3cfff53/R/snp_call.R#L208

**Comment by matt-gust** [comment]
I removed the lines in make_loom_files.sh that optionally created the conda environment since I'm already making it with a system() call in the R function.

I also fixed the paths to make_environment.yml and make_loom_files.sh in the r_make_loom_files function to reflect them being dumped a folder up from inst

File: inst/make_loom_files.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+#SBATCH --account=placeholder_account
+#SBATCH --output=placeholder_slurm_out
+#SBATCH --error=placeholder_slurm_out
+#SBATCH --job-name rrr_make_loom_files
+#SBATCH --array=0-0
```
Comment (MVesuviusC) [inline inst/make_loom_files.sh]: If you're not doing array jobs you don't need this line

File: inst/make_loom_files.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#SBATCH --nodes=1
+#SBATCH --ntasks=1
+#SBATCH --cpus-per-task=4
+#SBATCH --mem=80G
```
Comment (MVesuviusC) [inline inst/make_loom_files.sh:10]: Check your slurm output to see how much ram you're actually using, double it and round up to a round number to figure out how much ram to use. Also check if you need four cpus

File: R/velocity_analysis_functions.R [hunk 4]
```diff
@@ -0,0 +1,276 @@
+#' Wrapper to make loom files in R
```
Comment (MVesuviusC) [inline R/velocity_analysis_functions.R]: The title here should be Title Case

File: inst/make_loom_files.sh (hunk 3 above)
Comment (matt-gust) [reply inst/make_loom_files.sh:10]: Max RSS was ~38000M for each job, doubling and rounding up comes out to 80G actually ¯\_(ツ)_/¯

**Comment by matt-gust** [comment]
I re-worked r_make_loom_files and make_loom_files.sh to be an array job, that way the loom files don't have to be created sequentially

**Review by MVesuviusC — APPROVED** [review]
All good, yo

**Review by MVesuviusC — APPROVED** [review]
I think we're good

**Pull request merged**